### PR TITLE
Return successful `retcode` on `MaxIters` for `OptimizationSolution`

### PR DESCRIPTION
**Pull request by Vaibhavdixit02**
For global, stochastic gradient etc maxiters is used as the stopping criteria

### CODE REVIEW

File: src/retcodes.jl [hunk 1]
```diff
@@ -437,3 +437,11 @@ function successful_retcode(retcode::ReturnCode.T)
         retcode == ReturnCode.FloatingPointLimit
 end
 successful_retcode(sol::AbstractSciMLSolution) = successful_retcode(sol.retcode)
+
+function successful_retcoe(sol::OptimizationSolution)
```
Comment (Vaibhavdixit02) [inline src/retcodes.jl:441]:
```suggestion
function successful_retcode(sol::OptimizationSolution)
```

File: src/retcodes.jl [hunk 2]
```diff
@@ -437,3 +437,11 @@ function successful_retcode(retcode::ReturnCode.T)
         retcode == ReturnCode.FloatingPointLimit
 end
 successful_retcode(sol::AbstractSciMLSolution) = successful_retcode(sol.retcode)
+
+function successful_retcode(sol::OptimizationSolution)
+    retcode == ReturnCode.Success || retcode == ReturnCode.Terminated ||
+        retcode == ReturnCode.ExactSolutionLeft ||
+        retcode == ReturnCode.ExactSolutionRight ||
+        retcode == ReturnCode.FloatingPointLimit || 
+        retcode == ReturnCode.MaxIters
```
Comment (sathvikbhagavan) [inline src/retcodes.jl:446]:
```suggestion
    retcode == successful_retcode(sol.retcode) || retcode == ReturnCode.MaxIters
```

File: src/retcodes.jl [hunk 3]
```diff
@@ -437,3 +437,7 @@ function successful_retcode(retcode::ReturnCode.T)
         retcode == ReturnCode.FloatingPointLimit
 end
 successful_retcode(sol::AbstractSciMLSolution) = successful_retcode(sol.retcode)
+
+function successful_retcode(sol::OptimizationSolution)
+    retcode == successful_retcode(sol.retcode) || retcode == ReturnCode.MaxIters
```
Comment (Vaibhavdixit02) [inline src/retcodes.jl:442]:
```suggestion
    successful_retcode(sol.retcode) || retcode == ReturnCode.MaxIters
```

File: src/retcodes.jl [hunk 4]
```diff
@@ -437,3 +437,7 @@ function successful_retcode(retcode::ReturnCode.T)
         retcode == ReturnCode.FloatingPointLimit
 end
 successful_retcode(sol::AbstractSciMLSolution) = successful_retcode(sol.retcode)
+
+function successful_retcode(sol::OptimizationSolution)
+    successful_retcode(sol.retcode) || retcode == ReturnCode.MaxIters
```
Comment (Vaibhavdixit02) [inline src/retcodes.jl:442]:
```suggestion
    successful_retcode(sol.retcode) || sol.retcode == ReturnCode.MaxIters
```

**Comment by ChrisRackauckas** [comment]
This is not the solution. The solution is to just correctly throw `Success` if that's what the return code is supposed to be. `MaxIters` is defined as a non-converged early max iterations return, not a successful return.

**Pull request closed without merging**